frontend/src/main/scala/bloop/io/DirChecksum.scala

```diff
@@ -0,0 +1,106 @@
+package bloop.io
+
+import java.io.{InputStream, SequenceInputStream}
+import java.nio.file.Files
+import java.nio.file.attribute.FileTime
+import java.util.Enumeration
+import java.util.zip.{Adler32, CheckedInputStream}
+
+/**
+ * Represents the state of a directory and some files in which we track changes.
+ *
+ * @param dir           The directory to track.
+ * @param pattern       The pattern matching the files that must be tracked inside `dir`.
+ * @param modifiedTimes The last modification time of the tracked files.
+ * @param contentsChecksum The checksum of all the contents of the directory.
+ */
+final class DirChecksum(dir: AbsolutePath,
```

Comment:
What about calling this `BuildSignature` or `BuildChecksum` instead of `DirChecksum`?

Reply:
It's not really related to `Build`, we just happen to use it for `Build`?

Reply:
We agreed on `FileTracker`